taskvine/src/manager/vine_schedule.c

```diff
@@ -86,6 +67,57 @@ int vine_schedule_in_ramp_down(struct vine_manager *q)
 	return 0;
 }
 
+/* Check if worker resources are enough to run the task.
+ * @param q     Manager info structure.
+ * @param w     Worker info structure.
+ * @param t     Task info structure.
+ * @param tr    Task resources.
+ * @return 1 if yes, 0 otherwise. */
+int check_worker_have_enough_resources(
```

Comment:
I like how you broke this out separately here.

Put in some big fat comments at the top here explaining that the check takes into account killing unneded libraries.